test/azure/scan.spec.ts

```diff
@@ -0,0 +1,61 @@
+import {expect, test} from '@oclif/test';
+import {resourceGroup, subscriptionId} from '.';
+
+describe('Scan Integration Tests', function () {
+  this.slow(4000);
+  this.timeout(6000);
+  test
+    .stdout()
+    .command([
+      'scan',
+      '--scope',
+      subscriptionId,
+      '--file',
+      '../test/rules.json',
+    ])
+    .it('runs scan --scope [subscriptionId] --file ../test/rules.json', ctx => {
+      expect(ctx.stdout).to.not.contain('    bad-query\n');
```

Comment:
We may want to consider our expect checks & make sure that we aren't making our tests too brittle or accidentally adding things that are likely to change

That is - if we check every line, character & indent: we may have to update our tests often as we change possibly-unrelated bits in our code / test suite. We may also get false positives. Think about what happens when we add a third resourceGraph rule to the `rules.json` file, or when when we deploy additional resources in the ARM template. We have options: just change the tests, split data for each test, assert on only the critical things that we care about at the top level while still having confidence in the content due to unit tests, etc.

I expect us to come touch a bunch of this again in the near future based on what I'm seeing here

Reply:
Yes, I wasn't sure about the best way to go about testing this. I assume that we will be adding more rules to the test/rules.json file and updating the ARM template as needed to test those individual rules. The current test rules will likely be removed and replaced with the real rules once we start implementing them. 

My goal was to test the formatting of the scan command and I wanted to have tests to go along with the code in this PR. Is there a better way to approach this?